kubernetes-ingress.md

```diff
@@ -0,0 +1,150 @@
+# Ingress
+[⬅️ Back to Kubernetes overview](README.md)
+
+
+## Setup ingress controller
+![Ingress Controller](./kubernetes/ingress.png)
+
+### When using kind:
+From Kind [Documentation](https://kind.sigs.k8s.io/docs/user/ingress/)
+1. Setup Cluster
+
+```sh
+kind delete cluster #if one is alrady created
+cat <<EOF | kind create cluster --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
+  extraPortMappings:
+  - containerPort: 80
+    hostPort: 80
+    protocol: TCP
+  - containerPort: 443
+    hostPort: 443
+    protocol: TCP
+EOF
+```
+> [!WARNING]
+> you may need to change the hostport when some other service is already using it
+
+2. Create Namespace
+
+```sh
+kubectl create namespace ingress-nginx
+```
+3. Deploy Ingress Controller. [Github link](https://github.com/kubernetes/ingress-nginx)
+```sh
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+```
+
+4. Run this command to wait for ingress to finish setting up
+```sh
+kubectl wait --namespace ingress-nginx \
+  --for=condition=ready pod \
+  --selector=app.kubernetes.io/component=controller \
+  --timeout=90s
+```
+
+### When using Minikube
+From the setup guide in the [Documentation](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/) 
+
+```sh
+minikube addons enable ingress
+minikube service -n ingress-nginx ingress-nginx-controller
+```
+
+## Setup applications 
+
+1. Deploy a hello world app:
+```sh
+kubectl create deployment web --image=gcr.io/google-samples/hello-app:1.0
+kubectl expose deployment web --type=NodePort --port=8080
```

Comment:
Have we used this yet in the documentation? If not, we should explain that `expose` actually just creates a service.